[#239] Suggest storing the contract metadata on IPFS (#253)
tqtezos/baseDAO: docs/building.md

to generate the contract as well as all the example DAOs' initial storage
defined below.

## Generating a contract metadata

BaseDAO metadata should be compliant with [TZIP-16](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md).

> To print the metadata you'll need the `stack` tool, see

Reviewer: ```suggestion
You can generate the known metadata, but you'll need the `stack` tool installed, see
```

[The Haskell Tool Stack](https://docs.haskellstack.org/en/stable/README/) tutorial
for instructions on how to obtain it.

```bash
make metadata > metadata.json
```

> 

Reviewer: ```suggestion

You can then modify the produced `metadata.json` if you want to add more information.

```
I remembered now that we previously had a similar explanation as we only produce **known** parts of the metadata (hence the previous comment as well.

Author: I've push an update on this, as well as handle other changes. Have to update makefile a bit since its output was not as expected.

This metadata should be stored either:
- On IPFS: [Get started with IPFS](https://ipfs.io/)
- As a separate contract: [The `tezos-storage` URI Scheme](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#the-tezos-storage-uri-scheme)

To use this metadata, follow [TZIP-16 Contract storage](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#contract-storage)

Specifically:
> The encoding of the values must be the direct stream
of bytes of the data being stored. For instance, an URI starting with `http:`
will start with the 5 bytes `0x687474703a` (`h` is `0x68`, `t` is `0x74`,
etc.). There is no implicit conversion to Michelson's binary format (`PACK`) nor
quoting mechanism.

After the conversion, the bytes could be use in the argument `metadata_map` of the `make <storage>` command

> (See [Storing](#storing) section for more detail.)

Reviewer: Oh I see now that you mention the storing section here, however with the reordering you can point at the `Generating a contract storage` section or even just say "the next section", given that it immediately follows.

Following are the intructions to generate the initial storages of the DAO configuration

> provided in this codebase, see the [README](../README.md).

Reviewer: This seems off as there are no instructions "following" (they are after the `Generating a contract metadata` section now).

I suggest moving the new `Generating a contract metadata` section above this `Generating a contract storage` one instead.

Author: Make sense, I've updated this.

## Generating a contract metadata

BaseDAO metadata should be compliant with [TZIP-16](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md).

To print the metadata you'll need the `stack` tool, see
[The Haskell Tool Stack](https://docs.haskellstack.org/en/stable/README/) tutorial
for instructions on how to obtain it.

```bash
make metadata > metadata.json
```

This metadata should be stored either:
- On IPFS: [Get started with IPFS](https://ipfs.io/)
- As a separate contract: [The `tezos-storage` URI Scheme](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#the-tezos-storage-uri-scheme)

To use this metadata, follow [TZIP-16 Contract storage](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#contract-storage)

Specifically:

> > Unless otherwise-specified, the encoding of the values must be the direct stream

Reviewer: I think we can remove the "Unless otherwise-specified" as we don't specify a different format here (now we should), so IMO we can just cut the quote to:
```suggestion
> the encoding of the values must be the direct stream
```

Author: Updated.

Following are the intructions to generate the initial storages of the DAO configuration
provided in this codebase, see the [README](../README.md).

> ### Storing

Reviewer: I think this is unnecessary now, probably left after the previous change.
